src/main/scala/com/actian/spark_vector/colbuffer/time/TimeColumnBuffer.scala

```diff
@@ -15,43 +15,69 @@
  */
 package com.actian.spark_vector.colbuffer.time
 
+import org.apache.spark.sql.catalyst.util.DateTimeUtils
+
 import com.actian.spark_vector.colbuffer._
-import com.actian.spark_vector.colbuffer.util.TimeConversion
+import com.actian.spark_vector.colbuffer.util.{ TimeConversion, PowersOfTen, MillisecondsScale, NanosecondsScale }
 
 import java.nio.ByteBuffer
 import java.sql.Timestamp
+import java.util.Calendar
 
-private case class TimeColumnBufferParams(cbParams: ColumnBufferBuildParams, converter: TimeConversion.TimeConverter, adjustToUTC: Boolean = false)
+private case class TimeColumnBufferParams(cbParams: ColumnBufferBuildParams, converter: TimeConversion.TimeConverter,
+  adjustToUTC: Boolean = false)
 
 private[colbuffer] abstract class TimeColumnBuffer(p: TimeColumnBufferParams,  valueWidth: Int) extends
-  ColumnBuffer[Timestamp](p.cbParams.name, p.cbParams.maxValueCount, valueWidth, valueWidth, p.cbParams.nullable) {
-  override protected def put(source: Timestamp, buffer: ByteBuffer): Unit = {
+  ColumnBuffer[Timestamp, Long](p.cbParams.name, p.cbParams.maxValueCount, valueWidth, valueWidth, p.cbParams.nullable) {
+  private val ts = new Timestamp(System.currentTimeMillis())
+
+  override def put(source: Timestamp, buffer: ByteBuffer): Unit = {
     if (p.adjustToUTC) {
       TimeConversion.convertLocalTimestampToUTC(source)
     }
-    val convertedSource = p.converter.convert(TimeConversion.normalizedTime(source), p.cbParams.scale)
+    val convertedSource = p.converter.convert(TimeConversion.normalizeTime(source), p.cbParams.scale)
     putConverted(convertedSource, buffer)
   }
 
   protected def putConverted(converted: Long, buffer: ByteBuffer): Unit
+
+  override def get(buffer: ByteBuffer): Long = {
+    val deconvertedSource = p.converter.deconvert(getConverted(buffer), p.cbParams.scale)
+    ts.setTime(TimeConversion.scaleNanos(deconvertedSource, MillisecondsScale))
```

Comment:
shouldn't you pass here the cbParams.scale here? Or do you first scale it to milliseconds scale?